GANDLF/data/generate_calib_data.py

```diff
@@ -0,0 +1,240 @@
+            )
+
+            if params["medcam_enabled"]:
+                attention_map_aggregator = torchio.inference.GridAggregator(
+                    grid_sampler,
+                    overlap_mode=params["inference_mechanism"][
+                        "grid_aggregator_overlap"
+                    ],
+                )
+
+            current_patch = 0
+            for patches_batch in patch_loader:
+                if params["verbose"]:
+                    print(
+                        "=== Current patch:",
+                        current_patch,
+                        ", time : ",
+                        get_date_time(),
+                        ", location :",
+                        patches_batch[torchio.LOCATION],
+                        flush=True,
+                    )
+                current_patch += 1
+                image = (
+                    torch.cat(
+                        [
+                            patches_batch[key][torchio.DATA]
+                            for key in params["channel_keys"]
+                        ],
+                        dim=1,
+                    )
+                    .float()
+                    .to(params["device"])
+                )
+
+                # calculate metrics if ground truth is present
+                label = None
+                if params["problem_type"] != "segmentation":
+                    label = label_ground_truth
+                elif "label" in patches_batch:
+                    label = patches_batch["label"][torchio.DATA]
+
+                if label is not None:
+                    label = label.to(params["device"])
+                    if params["verbose"]:
+                        print(
+                            "=== Calibration shapes : label:",
+                            label.shape,
+                            ", image:",
+                            image.shape,
+                            flush=True,
+                        )
+                calib_image_dataset.append(image)
+                calib_label_dataset.append(label)
+
+
+    random.seed(params["calib_data_sample_seed"])
+    index = random.sample(range(len(calib_image_dataset)), int(params['calib_ratio'] * len(calib_image_dataset)))
+    
+    calib_image_sample = torch.Tensor([calib_image_dataset[i].numpy() for i in index])
+    calib_label_sample = torch.Tensor([calib_label_dataset[i].numpy() for i in index])
+
+    calib_dataset = TensorDataset(calib_image_sample, calib_label_sample)
+    
+    calib_dataloader =  DataLoader(dataset = calib_dataset, shuffle=False, batch_size=1)
+    return(model, calib_dataloader)
+
+
+if __name__ == "__main__":
+    import argparse, pickle, pandas
+
+    torch.multiprocessing.freeze_support()
+    # parse the cli arguments here
+    parser = argparse.ArgumentParser(description="NNCF POT of Trained Model")
+    parser.add_argument(
+        "-calib_data_csv", type=str, help="CSV file of the calibration data", required=True
```

Comment:
I wonder if csv format is something generic in GaNDLF? What kind of dataset can be plugged here this way?

Reply:
Yes. The data is preprocessed and csv files are generated for the training/testing/validation.